Add release notes for 1.7.2 (#399)
crystal-lang/crystal-website: _releases/2023-01-23-1.7.2-released.md

---
title: Crystal 1.7.2 is released!
version: 1.7.2
summary:
thumbnail: +
author:
---

We are releasing the second patch release of the [1.7](/2023/01/09/1.7.0-released/) series.

Pre-built packages are available on [GitHub Releases](https://github.com/crystal-lang/crystal/releases/tag/1.7.2) and our official distribution channels.
See [crystal-lang.org/install](https://crystal-lang.org/install/) for installation instructions.

This patch includes two bugfixes for regressions introduced in 1.7.0 and a minor UX bug in the compiler.

> The parser in 1.7.0 added more restrictions on what tokens can follow immediately after the name in a class definition. This was too restrictive and unintentionally affected code that should be considered valid ([#12977](https://github.com/crystal-lang/crystal/pull/12977)).

Reviewer: Maybe it's my sleepy self, but the way this paragraph is structured was hard to parse for me.

```suggestion

The first regression affected the parser, which in 1.7.0 was modified to restrict the set of tokens that can be placed after the name of a class in its definition. This restriction was however overly strict, and it unintentionally affected valid code ([#12977](https://github.com/crystal-lang/crystal/pull/12977)).
```

> Changes in 1.7.0 to an internal runtime method for loading debug info lead to a side-effect on large IO operations on some platforms. The exact reasons for this codegen bug are not entirely understood yet, but a workaround fixes the immediate effect.

Reviewer: ```suggestion

The second regression found in 1.7.0 affected the generated machine code in some platforms, where an internal runtime method for loading debug info lead to a side-effect on large IO operations. The exact reasons for [this codegen bug](https://github.com/crystal-lang/crystal/issues/12982) are not entirely understood yet, but [a workaround](https://github.com/crystal-lang/crystal/pull/12992) fixes the immediate effect.
```

> Finally, there's a fix for an annyoing compiler error predominantly appearing for Windows users which don't have a linker installed ([#12899](https://github.com/crystal-lang/crystal/pull/12899)).

Reviewer: ```suggestion

Finally, in this patch we included a fix for an unhelpful compiler error predominantly appearing for Windows users which don't have a linker installed ([#12899](https://github.com/crystal-lang/crystal/pull/12899)).
```
